src/commands/deploy/getOrCreateFunctionApp.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type IAppServiceWizardContext } from "@microsoft/vscode-azext-azureappservice";
+import { AzureWizard, nonNullProp, type ISubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { l10n } from "vscode";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+import { ResolvedFunctionAppResource } from "../../tree/ResolvedFunctionAppResource";
+import { type SlotTreeItem } from "../../tree/SlotTreeItem";
+import { SubscriptionListStep } from "../SubscriptionListStep";
+import { type IFunctionAppWizardContext } from "../createFunctionApp/IFunctionAppWizardContext";
+import { FunctionAppListStep } from "./FunctionAppListStep";
+import { type IFuncDeployContext } from "./deploy";
+
+export async function getOrCreateFunctionApp(context: IFuncDeployContext & Partial<IFunctionAppWizardContext>): Promise<SlotTreeItem> {
+    let node: SlotTreeItem | undefined;
+
+    const promptSteps = [new SubscriptionListStep(), new FunctionAppListStep()];
+    const title: string = l10n.t('Select Function App');
+    const wizard: AzureWizard<IAppServiceWizardContext> = new AzureWizard(context, {
+        promptSteps,
+        title
+    });
+
+    await wizard.prompt();
+
+    node = context.site ? await ext.rgApi.tree.findTreeItem(nonNullProp(context.site, 'id'), context) : undefined;
```

Comment:
Find tree item is sort of sketchy to use. Have you tested this when the tree hasn't been loaded yet?

Reply:
Yeah, it works though it's pretty unbearably slow. I may try to refactor this to just use the data model instead because of the performance.

Reply:
It's pretty tough without some significant refactoring. We need to pass the tree ndoe to deploy because it does the whole temporary description thing on it. If the node isn't there or if the parent is undefined, it'll throw an error.